src/goto-programs/goto_trace.cpp

```diff
@@ -104,9 +104,42 @@ void goto_trace_stept::output(
   out << '\n';
 }
 
-std::string trace_value_binary(
+static std::string
+numeric_representation(const exprt &expr, const trace_optionst &options)
+{
+  std::string result;
+  std::string prefix;
+  if(options.hex_representation)
+  {
+    mp_integer value_int =
+      binary2integer(id2string(to_constant_expr(expr).get_value()), false);
+    result = integer2string(value_int, 16);
+    prefix = "0x";
+  }
+  else
+  {
+    prefix = "0b";
+    result = expr.get_string(ID_value);
+  }
+
+  std::ostringstream oss;
+  std::string::size_type i = 0;
+  for(const auto c : result)
+  {
+    oss << c;
+    if(++i % 8 == 0 && result.size() != i)
+      oss << ' ';
+  }
+  if(options.base_prefix)
+    return prefix + oss.str();
+  else
+    return oss.str();
+}
+
+std::string trace_value_lowlevel(
   const exprt &expr,
```

Comment:
This could use a Doxy comment explaining what format you'll get and the customisations available via `options`